core/src/io/anuke/mindustry/world/blocks/power/PowerDiode.java

```diff
@@ -0,0 +1,87 @@
+package io.anuke.mindustry.world.blocks.power;
+
+import io.anuke.arc.Core;
+import io.anuke.mindustry.ui.Bar;
+import io.anuke.arc.util.Eachable;
+import io.anuke.mindustry.ui.Cicon;
+import io.anuke.mindustry.world.Tile;
+import io.anuke.mindustry.world.Block;
+import io.anuke.arc.graphics.g2d.Draw;
+import io.anuke.mindustry.graphics.Pal;
+import io.anuke.arc.graphics.g2d.TextureRegion;
+import io.anuke.mindustry.entities.traits.BuilderTrait;
+
+public class PowerDiode extends Block{
+
+    protected TextureRegion arrow;
+
+    public PowerDiode(String name){
+        super(name);
+        rotate = true;
+        update = true;
+        solid = true;
+        insulated = true;
+    }
+
+    @Override
+    public void update(Tile tile){
+        super.update(tile);
+
+        if(!tile.back().block().hasPower || !tile.front().block().hasPower) return;
+
+        PowerGraph backGraph = tile.back().entity.power.graph;
+        PowerGraph frontGraph = tile.front().entity.power.graph;
+        if(backGraph == frontGraph) return;
+
+        // 0f - 1f of battery capacity in use
+        Float backStored = backGraph.getBatteryStored() / backGraph.getTotalBatteryCapacity();
```

Comment:
Should be `float`, not Float.

Reply:
Weird how neither me nor my ide noticed that.